meson.build

```diff
@@ -42,7 +42,7 @@ if m_dep.found()
   add_project_link_arguments('-lm', language : 'c')
 endif
 
-tempita = files('sklearn/_build_utils/tempita.py')
+tempita = files('tempita.py')
```

Comment:
I had to do this to avoid breaking the `sklearn` import. Since Cython is only a build-time dependency, I suggest keeping this file outside the `sklearn/` folder to avoid having to import it in Pyodide and outside it, as described in https://github.com/scikit-learn/scikit-learn/pull/29791#issuecomment-2333746467 (even though, Cython is available as a pure Python wheel, we shouldn't require it).

Reply:
This is weird, the pattern was copied from `numpy` I think. Also this seems to imply that there are some differences in the test discovery between the js script and the pyodide venv ...

Reply:
As for the `tempita` pattern: NumPy indeed uses the same one, which apparently means that the `sklearn/_build_utils/` files are included in the scikit-learn sdist (and subsequently, wheel), but NumPy's are not. I've traced this to the use of how tempita is found in NumPy vs scikit-learn – we use [`files()`](https://mesonbuild.com/Reference-manual_functions.html#files) here, but NumPy uses [`find_program()`](https://mesonbuild.com/Reference-manual_functions.html#find_program). Let's hope 3af4ea294ead7570c665ea72620d77f6e021daf5 performs the desired fix.

Reply:
Argh, I'd like to do this in a separate PR to make the diff smaller... just pushed a9d7e6cd1c080dca9fd362dd8d14fbf1b6c2f996. Let's see if this works.

Reply:
Okay, got it working with a few tries. Some further cleanups could be done, but not too relevant here.

Reply:
> Argh, I'd like to do this in a separate PR to make the diff smaller... just pushed https://github.com/scikit-learn/scikit-learn/commit/a9d7e6cd1c080dca9fd362dd8d14fbf1b6c2f996. Let's see if this works.

Thanks, keeping the diff small is always appreciated! Removing `sklearn/_build_utils` from the wheel makes sense.

> This is weird, the pattern was copied from numpy I think. Also this seems to imply that there are some differences in the test discovery between the js script and the pyodide venv ...

I think the difference is that `--doctest-modules` is used in the pyodide venv runner because the local folder is mounted correctly and `setup.cfg` is seen (`setup.cfg` has `--doctest-modules`). Without `--doctest-modules`, `sklearn/_build_utils/tempita.py` is never imported.